engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTest.java

```diff
@@ -487,4 +488,31 @@ public void shouldCompletedProcessInstanceWhenElementsInsideExistingFlowScopesAr
         .describedAs("Expect the process instance to have been completed")
         .isPresent();
   }
+
+  @Test
+  public void shouldRejectCommandWhenProcessInstanceIsUnknown() {
+    // given
+    final long unknownKey = 12345L;
+
+    // when
+    ENGINE
+        .processInstance()
+        .withInstanceKey(unknownKey)
+        .modification()
+        .activateElement("A")
+        .expectRejection()
+        .modify();
```

Comment:
❌ We should also verify the correct client response

Reply:
❓ I'm sure what you mean. We test all the processor rejections like this.

Reply:
My intention was that writing a command rejection to the log is something different from writing a rejection as a response to the client.

🤔 I thought the `modify` method would return what is written to the response writer, but that's not the case. It simply also uses the recording exporter to retrieve the command rejection that was written to the log. 🤷 

@saig0 Are you aware of any tests that verify the rejection responses written using the responsewriter?

Reply:
@korthout the only test in the engine with response verification is [CreateProcessInstanceWithResultTest](https://github.com/camunda/zeebe/blob/main/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceWithResultTest.java) 

Otherwise, we verify the response in the integration tests, for example: [CreateProcessInstanceTest](https://github.com/camunda/zeebe/blob/main/qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/command/CreateProcessInstanceTest.java#L176-L192)